verification/src/types/unexpected.rs

```diff
@@ -0,0 +1,72 @@
+//! Error utils
+
+use std::fmt;
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+/// Error indicating an expected value was not found.
+pub struct Mismatch<T> {
```

Comment:
why not just two separate errors? 
`Mismatch` and `MismatchFound`

Reply:
Why to create different types while using just one would be enough to successfully do what it is supposed to and being clear about internal functionality and usage?